admin/load.php

```diff
@@ -503,3 +508,81 @@ function perflab_dismiss_wp_pointer_wrapper() {
 	check_ajax_referer( 'dismiss_pointer' );
 }
 add_action( 'wp_ajax_dismiss-wp-pointer', 'perflab_dismiss_wp_pointer_wrapper', 0 );
+
+/**
+ * Callback function to handle admin scripts.
+ *
+ * @since n.e.x.t
+ */
+function perflab_enqueue_modules_page_scripts() {
+	wp_enqueue_script( 'updates' );
+
+	wp_localize_script(
+		'updates',
+		'_wpUpdatesItemCounts',
+		array(
+			'settings' => array(
+				'totals' => wp_get_update_data(),
+			),
+		)
+	);
+
+	wp_enqueue_script( 'thickbox' );
+	wp_enqueue_style( 'thickbox' );
+}
+
+/**
+ * Callback function hooked to admin_action_perflab_activate_plugin to handle plugin activation.
+ *
+ * @since n.e.x.t
+ */
+function perflab_activate_plugin() {
+	// Do not proceed if plugin query arg is not present.
+	if ( empty( $_GET['plugin'] ) ) {
+		exit;
+	}
+
+	// The plugin being activated.
+	$plugin = filter_input( INPUT_GET, 'plugin' );
+
+	if ( check_admin_referer( "perflab_activate_plugin_{$plugin}" ) && current_user_can( 'activate_plugin', $plugin ) ) {
+		// Activate the plugin in question and return to prior screen.
+		activate_plugins( $plugin );
+		$referer = wp_get_referer();
+		if ( wp_safe_redirect( $referer ) ) {
+			exit;
+		}
+	} else {
+		exit;
+	}
+}
+
+add_action( 'admin_action_perflab_activate_plugin', 'perflab_activate_plugin' );
+
+/**
+ * Callback function hooked to admin_action_perflab_deactivate_plugin to handle plugin deactivation.
+ *
+ * @since n.e.x.t
+ */
+function perflab_deactivate_plugin() {
+	// Do not proceed if plugin query arg is not present.
+	if ( empty( $_GET['plugin'] ) ) {
+		exit;
+	}
+
+	// The plugin being deactivated.
+	$plugin = filter_input( INPUT_GET, 'plugin' );
+
+	if ( check_admin_referer( "perflab_deactivate_plugin_{$plugin}" ) && current_user_can( 'deactivate_plugin', $plugin ) ) {
+		// Deactivate the plugin in question and return to prior screen.
+		deactivate_plugins( $plugin );
+		$referer = wp_get_referer();
+		if ( wp_safe_redirect( $referer ) ) {
+			exit;
+		}
+	} else {
+		exit;
```

Comment:
For testability, could this just be a `return`?

Reply:
@swissspidy seems a return here would have been completely unnecessary, so I have removed the else condition entirely, please see https://github.com/WordPress/performance/pull/864/commits/0a20c0a59cc2b027d56f803cf7d8d3f2f787102d